cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -85,9 +92,25 @@ type nameNamespace struct {
 	namespace string
 }
 
+// newKube2Sky creates a new instance of Kubernetes to SkyDNS bridge
+func newKube2Sky(config *etcdstorage.EtcdConfig, domain string, mutationTimeout time.Duration) (ks *kube2sky, err error) {
+	ks = &kube2sky{
+		domain:              domain,
+		etcdMutationTimeout: mutationTimeout,
+		quorum:              config.Quorum,
+	}
+
+	var etcdClient etcd.Client
+	if etcdClient, err = newEtcdClient(config); err != nil {
+		return nil, fmt.Errorf("Failed to create etcd client: %v", err)
+	}
+	ks.etcdKeysAPI = etcd.NewKeysAPI(etcdClient)
```

Comment:
You should probably hold onto the etcdClient in the struct for completeness.  IIRC there used to be a members check in the kube2sky code.